libraries/app/database_api.cpp

```diff
@@ -2140,18 +2141,19 @@ vector<withdraw_permission_object> database_api_impl::get_withdraw_permissions_b
    return result;
 }
 
-vector<withdraw_permission_object> database_api::get_withdraw_permissions_by_recipient(account_id_type account, withdraw_permission_id_type start, uint32_t limit)const
+vector<withdraw_permission_object> database_api::get_withdraw_permissions_by_recipient(const std::string account_id_or_name, withdraw_permission_id_type start, uint32_t limit)const
 {
-   return my->get_withdraw_permissions_by_recipient( account, start, limit );
+   return my->get_withdraw_permissions_by_recipient( account_id_or_name, start, limit );
 }
 
-vector<withdraw_permission_object> database_api_impl::get_withdraw_permissions_by_recipient(account_id_type account, withdraw_permission_id_type start, uint32_t limit)const
+vector<withdraw_permission_object> database_api_impl::get_withdraw_permissions_by_recipient(const std::string account_id_or_name, withdraw_permission_id_type start, uint32_t limit)const
 {
    FC_ASSERT( limit <= 101 );
    vector<withdraw_permission_object> result;
 
    const auto& withdraw_idx = _db.get_index_type<withdraw_permission_index>().indices().get<by_authorized>();
    auto withdraw_index_end = withdraw_idx.end();
+   const account_id_type account = get_account_from_string(account_id_or_name)->id; // ask this, maybe not safe
```

Comment:
If the account is not found, get_account_from_string throws.

Reply:
thank you, removed the comments from code.